src/core_memusage.h

```diff
@@ -33,6 +41,9 @@ static inline size_t RecursiveDynamicUsage(const CTransaction& tx) {
     for (std::vector<CTxOut>::const_iterator it = tx.vout.begin(); it != tx.vout.end(); it++) {
         mem += RecursiveDynamicUsage(*it);
     }
+    mem += tx.vJoinSplit.size() * JOINSPLIT_SIZE(tx.nVersion);
+    mem += tx.vShieldedOutput.size() * OUTPUTDESCRIPTION_SIZE;
+    mem += tx.vShieldedSpend.size() * SPENDDESCRIPTION_SIZE;
```

Comment:
This should use `memusage::DynamicUsage(tx.vJoinSplit)` etc, since what we care about here is the memory usage (which includes `malloc` overhead) rather than the serialized size. I think this also avoids the complexity of `JOINSPLIT_SIZE`, as `boost::variant` itself is [sized to fit its largest variant](https://stackoverflow.com/a/26154914), and all variants within `SproutProof` are fixed-size; thus the `sizeof(T)` in `DynamicUsage` should be sufficient.